### PR TITLE
fix errors from new flake8-comprehensions

### PR DESCRIPTION
**Pull request by chriselion**
A new version of flake8-comprehensions came out, and flagged some existing code. Fixes all the flagged parts.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repos:
 -   repo: https://github.com/python/black
-    rev: 19.3b0
+    rev: 19.10b0
```
Comment (chriselion) [inline .pre-commit-config.yaml]: I ran `pre-commit --autoupdate` to get latest versions of the hooks too.

File: ml-agents/mlagents/trainers/bc/models.py [hunk 2]
```diff
@@ -55,7 +55,11 @@ def __init__(
             self.action_masks = tf.placeholder(
                 shape=[None, sum(self.act_size)], dtype=tf.float32, name="action_masks"
             )
-            self.sample_action_float, _, normalized_logits = self.create_discrete_action_masking_layer(
+            (
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/bc/models.py]: new version of `black` did this. Do not like.

File: ml-agents/mlagents/trainers/barracuda.py [hunk 3]
```diff
@@ -44,9 +44,8 @@ def parse_args(description, source_extension, help):
     parser.add_argument("--print-supported-ops", action="store_true")
     parser.add_argument("--verbose", action="store_true")
     args = parser.parse_args()
-    args.compress_f16 = (
-        False
-    )  # TEMP: disabled, until properly implemented parser.add_argument('-f16', '--compress-f16', action='store_true')
+    # TEMP: disabled, until properly implemented parser.add_argument('-f16', '--compress-f16', action='store_true')
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/barracuda.py]: Black tried to rearrange these, I put them on their own line for clarify

File: ml-agents/mlagents/trainers/tensorflow_to_barracuda.py [hunk 4]
```diff
@@ -1550,7 +1548,7 @@ def convert(
 
     # Convert
     o_model = barracuda.Model()
-    o_model.layers, o_input_shapes, o_model.tensors, o_model.memories, o_model.globals = process_model(
+    o_model.layers, o_input_shapes, o_model.tensors, o_model.memories, o_model.globals, = process_model(
```
Comment (harperj) [inline ml-agents/mlagents/trainers/tensorflow_to_barracuda.py]: Feel like the trailing comma for these is odd, but 🤷‍♂

Comment (chriselion) [reply ml-agents/mlagents/trainers/tensorflow_to_barracuda.py]: Oops. I manually reverted some of the changes that a new version of black added, but missed those. Cleaning up now.

**Comment by chriselion** [comment]
@harperj good call, I already had pylint disabled on the barracuda files (for similar reasons). I undid the changes and disabled flake8 on them.

**Comment by harperj** [comment]
All good now assuming tests are passing.

**Pull request merged**